### PR TITLE
[WIP] Server/3.2 install restructure

### PR DESCRIPTION
**Pull request by rosieyohannan**
Branch to restructure the server installation guide.

### CODE REVIEW

File: jekyll/_cci2/server-3-install.adoc [hunk 1]
```diff
@@ -14,7 +14,16 @@ Before you begin with the CircleCI server v3.x installation process, ensure all
 
 toc::[]
 
-## Step 1 - Install KOTS
+## Phase 2 - Core Services Installation
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:17]: This is already the main H1 heading of the doc so can be taken out

File: jekyll/_cci2/server-3-install.adoc [hunk 2]
```diff
@@ -14,7 +14,16 @@ Before you begin with the CircleCI server v3.x installation process, ensure all
...
+CircleCI server v3.x uses link:https://kots.io/[KOTS] from link:https://www.replicated.com/[Replicated] to manage and distribute server v3.x. KOTS is a kubectl plugin. To install the latest version, you can run 'curl https://kots.io/install | bash'
+
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:19]: This section already exists below

File: jekyll/_cci2/server-3-install.adoc [hunk 3]
```diff
@@ -14,7 +14,16 @@ Before you begin with the CircleCI server v3.x installation process, ensure all
 
 toc::[]
 
-## Step 1 - Install KOTS
+## Phase 2 - Core Services Installation
+CircleCI server v3.x uses link:https://kots.io/[KOTS] from link:https://www.replicated.com/[Replicated] to manage and distribute server v3.x. KOTS is a kubectl plugin. To install the latest version, you can run 'curl https://kots.io/install | bash'
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:18]: Also, we ask people to install KOTS in phase 1 so this is duplication I think

File: jekyll/_cci2/server-3-install-prerequisites.adoc [hunk 4]
```diff
@@ -45,98 +69,253 @@ CircleCI server users: +
...
-| {helmversion} or greater
-| Kubernetes Package Management
+==== GKE
+You can learn more about creating a GKE cluster https://cloud.google.com/kubernetes-engine/docs/how-to#creating-clusters[here]. 
 
-| https://kots.io/[Kots]
-| {kotsversion} or greater
-| Replicated Kubernetes Application Management
-|===
+CAUTION: Do not use an Autopilot cluster. CircleCI requires functionality that is not supported by GKE Autopilot.
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install-prerequisites.adoc:103]:
```suggestion
```

Comment (rosieyohannan) [reply jekyll/_cci2/server-3-install-prerequisites.adoc:103]: I already stuck this in at the top

File: jekyll/_cci2/server-3-install-prerequisites.adoc [hunk 5]
```diff
@@ -45,99 +68,310 @@ CircleCI server users: +
...
+=== Create a new GitHub OAuth app
+Registering and setting up a new GitHub OAuth app for CircleCI server allows for authorization control to your server installation using GitHub OAuth and for updates to GitHub projects/repos using build status information.
+
+. In your browser navigate to **your GitHub instance** > **Settings** > **Developer Settings** > **OAuth Apps** and click the **New OAuth App** button. 
+
+. Complete the following fields based on your planned installation: 
+** Homepage URL: The URL of your planned CircleCI installation.
+** Authorization callback URL: The authorization callback URL will be the URL of your planned CircleCI installation followed by /auth/github 
+
+=== Frontend TLS certificates
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install-prerequisites.adoc:142]: Would be good to link to this section from the KOTS admin console where it says "see documentation"
Screenshot 2021-09-23 at 11 39 46
:

Comment (rosieyohannan) [reply jekyll/_cci2/server-3-install-prerequisites.adoc:142]: Also what happens if they uncheck this box - do we need further docs for this scenario?

File: jekyll/_cci2/server-3-install.adoc [hunk 6]
```diff
@@ -14,7 +14,16 @@ Before you begin with the CircleCI server v3.x installation process, ensure all
...
+
+Ensure you are running at least the minimum version (KOTS 1.47.3) by running `kubectl kots version`.
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:20]: Elsewhere I think we say KOTS 1.44.1 minimum, shall I use 1.47.3 instead?

Comment (rosieyohannan) [reply jekyll/_cci2/server-3-install.adoc]: @nathanfishcircleci

File: jekyll/_cci2/server-3-install-prerequisites.adoc (hunk 5 above)
Comment (rosieyohannan) [reply jekyll/_cci2/server-3-install-prerequisites.adoc:142]: @nathanfishcircleci

File: jekyll/_cci2/server-3-operator-externalizing-services.adoc [hunk 7]
```diff
@@ -70,11 +83,45 @@ Consider running at least two or more PostgreSQL replicas to enable recovery fro
...
+
+* *MongoDB authentication mechanism (required)* - 
+The authentication mechanism to use, usually SCRAM-SHA-1.
+
+* *Additional connection options (optional)* - 
+Any other connection options you would like to use. This needs to be formatted as a query string (`key=value` pairs, separated by `&``, special characters need to be URL encoded). See the link:https://docs.mongodb.com/v3.6/reference/connection-string/[MongoDB docs] for available options.
+
+== Vault
+
+Complete the following Vault fields:
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-operator-externalizing-services.adoc:127]: @nathanfishcircleci we need equivalents for Vault here I think

File: jekyll/_cci2/server-3-install.adoc (hunk 6 above)
Comment (nathanfishcircleci) [reply jekyll/_cci2/server-3-install.adoc]: yes. 1.47.3 is minimum. The most recent release is 1.50.3.

File: jekyll/_cci2/server-3-operator-externalizing-services.adoc (hunk 7 above)
Comment (nathanfishcircleci) [reply jekyll/_cci2/server-3-operator-externalizing-services.adoc:136]: Sorry!

Complete the following Vault fields: 

URL - The URL to your Vault service. 

Transit Path - Your Vault secrets transit path.

Token - The access token for vault.

**Comment by rosieyohannan** [comment]
OK I am going to merge these changes into the server 3.2 branch now so we can get things ready for release. If we need to make further changes to structure let's just open another branch @nathanfishcircleci .

**Pull request merged**